sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -356,6 +359,8 @@ abstract class UnaryExpression extends Expression {
       """
     }
   }
+
+  override def sql: Option[String] = child.sql.map(childSQL => s"($prettyName($childSQL))")
```

Comment:
Maybe it is good to add doc in the code to mention that `prettyName` should match the name registered in function registry.